core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -47,7 +50,9 @@
   INSTANCE;
 
   private BlockMasterClientPool mBlockMasterClientPool;
-  private BlockWorkerClientPool mLocalBlockWorkerClientPool;
+  /** A map from the worker's address to its client pool. */
+  private Map<WorkerNetAddress, BlockWorkerClientPool> mLocalBlockWorkerClientPoolMap =
+      new HashMap<>();
```

Comment:
It's thread safe, because the update is done in `initializeLocalBlockWorkerClientPool` which is already synchronized. But I changed it to `concurrentHashMap` to be extra safe.